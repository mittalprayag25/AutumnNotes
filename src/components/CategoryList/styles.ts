import {StyleSheet} from 'react-native';
import {sizes} from './../../constants/sizes';
import {colors} from './../../constants/colors';

const styles = StyleSheet.create({
  container: {
    //height: '20%',
  },
  header: {
    fontSize: sizes.font.md,
  },
  listItem: {
    height: 128,
    width: '45%',
    margin: sizes.space.sm,
    backgroundColor: colors.grey,
    alignItems: 'flex-start',
    padding: sizes.space.sm,
    borderRadius: sizes.borderRadius.sm,
  },
  categoryLabel: {
    fontSize: sizes.font.lg,
    marginLeft: sizes.space.sm,
    fontWeight: 'bold',
  },
  categoryGroup: {
    backgroundColor: colors.group,
    marginVertical: sizes.space.sm,
  },
});

export default styles;
