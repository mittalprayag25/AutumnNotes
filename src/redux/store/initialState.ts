const initialState = {
  notes: {
    allNotes: [],
    singleNote: null,
  },
  categories: {
    categories: [],
  },
};

export default initialState;
