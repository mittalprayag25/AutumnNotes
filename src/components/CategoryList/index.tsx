import React from 'react';
import {FlatList, Text, TouchableOpacity, View} from 'react-native';
import {Routes} from '../../constants/NavigationUtils';
import styles from './styles';

const CategoryList = ({navigation, data}: any) => {
  const renderItem = ({item, index}) => {
    return (
      <TouchableOpacity
        style={styles.listItem}
        testID={`note-detail-item-${item.title}`}
        onPress={() =>
          navigation.navigate(Routes.NOTEDETAIL, {
            singleNote: item,
            noteList: [],
            index: index,
          })
        }>
        <Text style={styles.header}>{`${item.title}`}</Text>
      </TouchableOpacity>
    );
  };

  return (
    <View style={styles.categoryGroup}>
      <Text style={styles.categoryLabel}>
        {data.category.length > 0 ? data.category : 'Misc'}
      </Text>
      <FlatList
        style={styles.container}
        data={data.data}
        renderItem={renderItem}
        numColumns={2}
      />
    </View>
  );
};

export default CategoryList;
