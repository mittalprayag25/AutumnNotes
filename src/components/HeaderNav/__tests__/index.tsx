import {act, fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import * as reactRedux from 'react-redux';
import {Provider} from 'react-redux';

import {
  createMockStore,
  mockNavigationWithParams,
} from '../../../__mocks__/index';
import CategoryList from '../index';
import initialState from './../../../redux/store/initialState';

const mockedDispatch = jest.fn();

describe('Home with success data', () => {
  const navigation = mockNavigationWithParams({});
  const props = {
    navigation: navigation,
    data: {
      category: 'category',
      data: [
        {category: 'category', title: 'title', description: 'description'},
      ],
    },
  };

  const changeState = {
    ...initialState,
  };
  it('should match snapshot', () => {
    const wrapper = render(<CategoryList {...props} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  // it('Should contain CLick More Text', () => {
  //   const {getByTestId} = render(
  //     <Provider store={createMockStore(changeState)}>
  //       <Home {...props} />
  //     </Provider>,
  //   );
  //   expect(getByTestId('more-records').props.children).toBe(
  //     'Click here for More stories.....',
  //   );
  // });
  // it('Should navigate to details', async () => {
  //   const useDispatchSpy = jest.spyOn(reactRedux, 'useDispatch');
  //   const mockedDispatch = jest.fn();
  //   useDispatchSpy.mockReturnValue(mockedDispatch);

  //   const {getByTestId} = render(
  //     <Provider store={createMockStore(changeState)}>
  //       <Home {...props} />
  //     </Provider>,
  //   );
  //   const scrollView = getByTestId('home-story-view');
  //   act(() => {
  //     fireEvent(getByTestId('more-records'), 'press');
  //   });
  //   expect(scrollView).toBeDefined();
  //   expect(mockedDispatch).toHaveBeenCalled();

  //   const {refreshControl} = scrollView.props;
  //   act(() => {
  //     refreshControl.props.onRefresh();
  //   });
  //   expect(mockedDispatch).toHaveBeenCalled();
  // });
});
