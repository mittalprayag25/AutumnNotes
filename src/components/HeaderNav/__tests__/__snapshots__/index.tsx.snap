// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home with success data should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 1,
      "borderColor": "#abc",
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "space-between",
      "paddingHorizontal": 8,
      "width": "100%",
    }
  }
/>
`;
